src/sqlfluff/dialects/dialect_greenplum.py

```diff
@@ -0,0 +1,181 @@
+                            ),
+                        ),
+                    )
+                ),
+                Sequence(
+                    "INHERITS",
+                    Bracketed(Delimited(Ref("TableReferenceSegment"))),
+                    optional=True,
+                ),
+            ),
+            # Create OF syntax:
+            Sequence(
+                "OF",
+                Ref("ParameterNameSegment"),
+                Bracketed(
+                    Delimited(
+                        Sequence(
+                            Ref("ColumnReferenceSegment"),
+                            Sequence("WITH", "OPTIONS", optional=True),
+                            AnyNumberOf(Ref("ColumnConstraintSegment")),
+                        ),
+                        Ref("TableConstraintSegment"),
+                    ),
+                    optional=True,
+                ),
+            ),
+            # Create PARTITION OF syntax
+            Sequence(
+                "PARTITION",
+                "OF",
+                Ref("TableReferenceSegment"),
+                Bracketed(
+                    Delimited(
+                        Sequence(
+                            Ref("ColumnReferenceSegment"),
+                            Sequence("WITH", "OPTIONS", optional=True),
+                            AnyNumberOf(Ref("ColumnConstraintSegment")),
+                        ),
+                        Ref("TableConstraintSegment"),
+                    ),
+                    optional=True,
+                ),
+                OneOf(
+                    Sequence("FOR", "VALUES", Ref("PartitionBoundSpecSegment")),
+                    "DEFAULT",
+                ),
+            ),
+        ),
+        AnyNumberOf(
+            Sequence(
+                "PARTITION",
+                "BY",
+                OneOf("RANGE", "LIST", "HASH"),
+                Bracketed(
+                    AnyNumberOf(
+                        Delimited(
+                            Sequence(
+                                OneOf(
+                                    Ref("ColumnReferenceSegment"),
+                                    Ref("FunctionSegment"),
+                                ),
+                                AnyNumberOf(
+                                    Sequence(
+                                        "COLLATE",
+                                        Ref("QuotedLiteralSegment"),
+                                        optional=True,
+                                    ),
+                                    Ref("ParameterNameSegment", optional=True),
+                                ),
+                            ),
+                        )
+                    )
+                ),
+            ),
+            Sequence("USING", Ref("ParameterNameSegment")),
+            OneOf(
+                Sequence(
+                    "WITH",
+                    Bracketed(
+                        Delimited(
```

Comment:
This is also different from Postgres but I think you're actually right here. Do you fancy correcting Postgres as part of this PR?

Reply:
Nah, think it's small enough (and obviously wrong enough) that it's OK. I added it as a comment above in the "Are there any other side effects of this change that we should be aware of?" section above.